src/Core/Utilities/ThemeManager.cs

```diff
@@ -147,16 +150,40 @@ public static string GetAutoDarkTheme()
             return stateService.GetAutoDarkThemeAsync().GetAwaiter().GetResult();
         }
 
+        //HACK: OsDarkModeEnabled() is divided into Android and iOS implementations due to a MAUI bug.
+        // Currently on iOS when resuming the app after showing a System "Share/Sheet" (or other similar UI)
+        // MAUI reports the incorrect Theme. To avoid this we are fetching the current OS Theme directly on iOS from the iOS API.
+        // MAUI Issue: https://github.com/dotnet/maui/issues/19614
         public static bool OsDarkModeEnabled()
         {
-            if (Application.Current == null)
-            {
-                // called from iOS extension
-                var app = new App(new AppOptions { IosExtension = true });
-                return app.RequestedTheme == AppTheme.Dark;
-            }
+#if ANDROID
             return Application.Current.RequestedTheme == AppTheme.Dark;
+#elif IOS
+            var requestedTheme = AppTheme.Unspecified;
+            if ((OperatingSystem.IsIOS() && !OperatingSystem.IsIOSVersionAtLeast(13, 0)))
+                return false;
+
+            var traits = InvokeOnMainThread(() => WindowStateManager.Default.GetCurrentUIViewController()?.TraitCollection) ?? UITraitCollection.CurrentTraitCollection;
```

Comment:
⛏️ If possible, use `#else` given that if it's not Android then it's for sure iOS and we avoid building warnings and check failures of Checkmarx on Github Actions.
⛏️ Also,  you're already in the iOS block of code so `OperatingSystem.IsIOS()` is not necessary.

Reply:
@fedemkr Agreed, changes done.